express-api/src/services/buildings/buildingServices.ts

```diff
@@ -0,0 +1,121 @@
+import { Building } from '@/typeorm/Entities/Building';
+import { AppDataSource } from '@/appDataSource';
+import { ErrorWithCode } from '@/utilities/customErrors/ErrorWithCode';
+import { DeepPartial, FindOptionsOrder } from 'typeorm';
+import { BuildingFilter } from '@/services/buildings/buildingSchema';
+
+const buildingRepo = AppDataSource.getRepository(Building);
+
+/**
+ * @description          Adds a new building to the datasource.
+ * @param   building     Incoming building data to be added to the database.
+ * @returns {Response}   A 201 status and the building data added.
+ * @throws ErrorWithCode If the building already exists or is unable to be added.
+ */
+export const addBuilding = async (building: Building) => {
+  const existingBuilding = await getBuildingById(building.Id);
+  if (existingBuilding) {
+    throw new ErrorWithCode('Building already exists.', 409);
+  }
+  const newBuilding = buildingRepo.save(building);
```

Comment:
I think .save returns a promise, so let's await it.
```suggestion
  const newBuilding = await buildingRepo.save(building);
```
Same issues exists with addParcel. Going to fix those and add to this PR as we go.

Reply:
Alternatively, directly return a Promise (ie. `return buildingRepo.save(building);`). The caller should be awaiting the Promise of this async function anyways.